src/main/pages/user-management/user-management.adoc

```diff
@@ -8,67 +8,102 @@ folder: user-management
+To find your Keycloak URL:
 
-[id="che-and-keycloak"]
-== Che and Keycloak
+If Che is deployed on Openshift:
 
-When Che is deployed on OpenShift or installed on Docker, either deployment script or CLI makes sure that Keycloak is properly configured. When a `che-public` client is created, there are two important fields to be populated:
+* Go to OpenShift web console > keycloak namespace 
 
-* *Valid Redirect URIs* need to be the URL you use to access Che. For example, if your original `CHE_HOST` is `1.1.1.1` but you access Che at `myhost` which is an alias, you will see an error from Keycloak, saying that it’s an invalid `redirectURI`. So, if this error occurs, go to Keycloak administration console and check valid Redirect URIs.
-* *Web Origins* - the same concerns Web Origins. An invalid web origin causes CORS error.
+If Che is running on Docker:
+ 
+* Go to `$CHE_HOST:5050/auth`   
 
-[id="keycloak-tokens"]
-== Keycloak Tokens
+[id="configuring-che-to-work-with-keycloak"]
+== Configuring Che to work with Keycloak
 
-By default, a user token gets expired after 30 mins. Keycloak admin can fine tune different realm settings related to Keycloak tokens:
+The deployment script ensures that Keycloak is properly configured when Che is deployed on OpenShift or installed on Docker. When the `che-public` client is created, the following fields are populated:
+
+* *Valid Redirect URIs*: Use this URL to access Che. 
+* *Web Origins*
+
+The following are common errors when configuring Che to work with Keycloak: 
+
+*Invalid `redirectURI` error*:  occurs when you access Che at `myhost`, which is an alias, and your 
+original `CHE_HOST` is `1.1.1.1`.  If this error occurs, go to the Keycloak administration console and the valid redirect URIs are configured.
+
+*CORS error*: occurs when you have an invalid web origin
+
+[id="configuring-keycloak-tokens"]
+== Configuring Keycloak tokens
+
+A user token expires after 30 minutes by default. 
+
+You can change the following Keycloak token settings:
 
 image::keycloak/keycloak_realm.png[]
 
-[id="user-federation"]
-== User Federation
+[id="setting-up-user-federation"]
+== Setting up user federation
+
+Keycloak federates external user databases and supports LDAP and Active Directory. You can test the connection and authenticate users before choosing a storage provider.
 
-Keycloak provides a user friendly page to http://www.keycloak.org/docs/3.2/server_admin/topics/user-federation.html[connect LDAP/Active Directory]. There are a number of http://www.keycloak.org/docs/3,2/server_admin/topics/user-federation/ldap.html[fields to be populated] on the config page, and those are specific to your particular LDAP instance, user filters, preferable mode etc. It is possible to test connection and authentication even before saving any particular storage provider.
+See http://www.keycloak.org/docs/3.2/server_admin/topics/user-federation.html[user storage federation] to learn how to add a provider.
```

Comment:
It's worth mentioning that this links to Keycloak documentation, not a section of the Che documentation.